README.md

```diff
@@ -100,6 +100,101 @@ You are free to implement the genome classes however you want, and using whateer
 
 When you have implemented the two (or more) classes, describe the complexity of each operation as a function of the genome size (at the time of the operation), and the size of the TE involved (and when copying, the offset you are copying). Put the description here:
 
-**FIXME: OPERATION COMPLEXITY**
+**Operation Complexity**
+
+n is the size of the genome
+m is the TE size
+l is the offset
+p is the number of active TE
+
+n > p, there can never be more active TE's than the length of the genome.
+
+### List Genome
+
+#### init
+
+init has a complexity of O(n).
+
+self.genome creates a list of length n in in time O(n), and is the most complex operation in the function.
+
+#### insert_te
+
+insert_te has a complecity of O(m*n)
+
+The most complex operation in the inser_te function is self.genome[pos:pos] = te. This has to insert the entire TE, moving every element after it by the length of the genome. Worst case would be inserting the TE at position 0, inserting m elements at zero (O(m)) and then pushing n elements by m (O(m*n)). Compared to this complexity, the key loop (O(p)) and the **pos < start** loop (O(p)) do not seem relevant. When a TE is inserted at 0, the **pos < start** loop has a max run-time of O(p) because every start position of every active TE must be updates. The **start < pos <= end** loop should evaluate p times, but only run once, as a new TE can only knock-out 1 other TE, giving it the O(p) complexity.
```

Comment:
If you inserted the new TE one nucleotide at a time, I would agree. But when you do `x[i:i] = val` it only takes time `len(x) + len(val)` so it is only $O(n+m)$. This still dominates running through all the $k$ TEs since $k\leq n$.